src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  health = 100;
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.health > 0 && !animal.hidden) {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      const position = Animal.alive.indexOf(animal);
+
+      Animal.alive.splice(position, 1);
```

Comment:
Using `indexOf` and `splice` within the `bite` method to remove dead animals is not performance-efficient as it involves nesting loops. Instead, use the `filter` method to remove dead animals from the `Animal.alive` array. This can be done in a separate method or directly within the `bite` method after adjusting the health of the animal.